IdentityCore/src/requests/MSIDInteractiveAuthorizationCodeRequest.m

```diff
@@ -160,25 +162,21 @@ - (void)getAuthCodeWithCompletionImpl:(MSIDInteractiveAuthorizationCodeCompletio
         }
         else if ([response isKindOfClass:MSIDWebOpenBrowserResponse.class])
         {
-            NSURL *browserURL = ((MSIDWebOpenBrowserResponse *)response).browserURL;
-
-#if TARGET_OS_IPHONE
-            if (![MSIDAppExtensionUtil isExecutingInAppExtension])
+            NSError *error = nil;
+            MSIDWebResponseBaseOperation *operation = [MSIDWebResponseOperationFactory createOperationForResponse:response error:&error];
+            if (error)
             {
-                MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"Opening a browser - %@", MSID_PII_LOG_MASKABLE(browserURL));
-                [MSIDAppExtensionUtil sharedApplicationOpenURL:browserURL];
-            }
-            else
-            {
-                NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorAttemptToOpenURLFromExtension, @"unable to redirect to browser from extension", nil, nil, nil, self.requestParameters.correlationId, nil, YES);
                 returnErrorBlock(error);
                 return;
             }
-#else
-            [[NSWorkspace sharedWorkspace] openURL:browserURL];
-#endif
-            NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorSessionCanceledProgrammatically, @"Authorization session was cancelled programatically.", nil, nil, nil, self.requestParameters.correlationId, nil, YES);
-            returnErrorBlock(error);
+            
+            [operation doActionWithCorrelationId:self.requestParameters.correlationId
+                                      completion:^(MSIDTokenResult * _Nullable __unused result, NSError * _Nullable error) {
+                if (error)
```

Comment:
why is result unused here?

Reply:
1. Based on what I knew, opening browser flow doesn't provide any call back or give any response in the end within the same flow. Will redirect back to the app later on from browser. I might miss some other flow, please correct me.
2. This was a bad  API design, I have updated it by passing in an error pointer only instead of using a call back block